celery/extractions.py

```diff
@@ -14,13 +20,109 @@
+
 taskmanager = Celery('extractions',
                      broker=CELERY_BROKER_URL,
                      backend=CELERY_RESULT_BACKEND)
 
+
+def run_command(args):
+    """
+    Run command specified by args. If exit code is not 0 then full command line, STDOUT, STDERR are printed and an
+    Exception is raised
+    :param args: array of argument
+    :return: None
+    """
+    p = Popen(args, stdout=PIPE, stderr=PIPE)
+    p.wait()
+
+    if p.returncode != 0:
+        print("Commande : %s" % " ".join(args))
+        print("Exit code : %s" % p.returncode)
+        print("STDOUT : %s" % p.stdout.read().decode())
+        print("STDERR : %s" % p.stderr.read().decode())
+        raise Exception("Error running %s" % " ".join(args))
+
+
+def get_all_tables(conn, year):
+    """
+    List tables in schema foncier_<year> where year is the second argument
+    :param conn: a psycopg connection instance
+    :param year: numerix year to append to 'foncier_' to build schema name
+    :return: array of table name
+    """
+    cur = conn.cursor()
+    cur.execute("SELECT table_name FROM information_schema.tables WHERE table_schema = 'foncier_%s'" % year)
+    res = [res[0] for res in cur.fetchall()]
+    cur.close()
+    return res
+
+
+def export_schema_to_shapefile_or_mapfile(year, proj, output_dir, format, conn, pg_connect_string):
+    """
+    Extract all table in schema foncier_<year> where 'year' is the first argument and generates files according to
+    'format' in 'output_dir' folder.
+    :param year: numeric year to append to 'foncier_' to build schema name
+    :param proj: output projection
+    :param output_dir: folder where files will be writen
+    :param format: ogr2ogr file format like : "ESRI Shapefile" or "MapInfo File"
+    :param conn: a psycopg connection instance
+    :param pg_connect_string: a string that contains option to connect to database with ogr2ogr. 'schema' option will
+    be added
+    :return: None
+    """
+    for table in get_all_tables(conn, year):
+        args = ["ogr2ogr", 
+                "-a_srs", "EPSG:%s" % proj,
+                "-t_srs", "EPSG:%s" % proj,
+                "-f", format, output_dir,
+                "PG:%s schemas=foncier_%s" % (PG_CONNECT_STRING, year),
+                table]
+        run_command(args)
+
+
+def export_schema_to_sql(year, proj, output_dir, conn, pg_connect_string):
+    """
+    Extract all table in schema foncier_<year> where 'year' is the first argument and one sql file to create schema,
+    tables and inserts data
+    :param year: numeric year to append to 'foncier_' to build schema name
+    :param proj: output projection
+    :param output_dir: folder where files will be writen
+    :param conn: a psycopg connection instance
+    :param pg_connect_string: a string that contains option to connect to database with ogr2ogr. 'schema' option will
+    be added
+    :return: None
+    """
+    with open(join(output_dir, "foncier_%s.sql"  % year), 'wb') as f:
```

Comment:
There are some UTF-8 characters that are easier to process with binary flag. I have to add this flag after testing against test data.